### PR TITLE
feat: improve location text

### PR DESCRIPTION
**Pull request by 0xMRTT**
See https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/issues/36 for more

### CODE REVIEW

File: src/location.c [hunk 1]
```diff
@@ -532,8 +532,8 @@ handle_start_in_thread_func (GTask *task,
 
       if (g_str_equal (app_id, ""))
         {
-          title = g_strdup (_("Grant Access to Your Location?"));
-          subtitle = g_strdup (_("An application wants to use your location."));
+          title = g_strdup (_("Allow Access to Your Location?"));
+          subtitle = g_strdup (_("An application wants to known the location of this device."));
```
Comment (smcv) [inline src/location.c:536]:
```suggestion
          subtitle = g_strdup (_("An application wants to know the location of this device."));
```

File: src/location.c [hunk 2]
```diff
@@ -545,11 +545,11 @@ handle_start_in_thread_func (GTask *task,
           info = g_desktop_app_info_new (id);
           name = g_app_info_get_display_name (G_APP_INFO (info));
 
-          title = g_strdup_printf (_("Give %s Access to Your Location?"), name);
+          title = g_strdup_printf (_("Allow %s to Access to Your Location?"), name);
```
Comment (smcv) [inline src/location.c:548]:
```suggestion
          title = g_strdup_printf (_("Allow %s to Access Your Location?"), name);
```

File: src/location.c [hunk 3]
```diff
@@ -545,11 +545,11 @@ handle_start_in_thread_func (GTask *task,
...
           if (g_desktop_app_info_has_key (info, "X-Geoclue-Reason"))
             subtitle = g_desktop_app_info_get_string (info, "X-Geoclue-Reason");
           else
-            subtitle = g_strdup_printf (_("%s wants to use your location."), name);
+            subtitle = g_strdup_printf (_("%s wants to known the location of this device."), name);
```
Comment (smcv) [inline src/location.c:552]:
```suggestion
            subtitle = g_strdup_printf (_("%s wants to know the location of this device."), name);
```

**Review by smcv — CHANGES_REQUESTED** [review]
(This is only an English correctness review, I have no particular opinion on whether the new wording is better/clearer.)

**Comment by smcv** [comment]
cc @allanday

File: po/cs.po [hunk 4]
```diff
@@ -116,21 +116,23 @@ msgid "Grant Access"
...
 #: src/location.c:535
-msgid "Grant Access to Your Location?"
+#, fuzzy
+msgid "Allow Access to Your Location?"
 msgstr "Schválit přístup k vaší poloze?"
 
 #: src/location.c:536
-msgid "An application wants to use your location."
+#, fuzzy
+msgid "An application wants to known the location of this device."
```
Comment (smcv) [inline po/cs.po:125]: The l10n stuff presumably needs to be regenerated for the updated string.

File: src/location.c [hunk 5]
```diff
@@ -545,11 +545,11 @@ handle_start_in_thread_func (GTask *task,
           info = g_desktop_app_info_new (id);
           name = g_app_info_get_display_name (G_APP_INFO (info));
 
-          title = g_strdup_printf (_("Give %s Access to Your Location?"), name);
+          title = g_strdup_printf (_("Allow %s to Access Your Location?"), name);
```
Comment (AlessandroBono) [inline src/location.c:548]: The latest [mockups](https://gitlab.gnome.org/Teams/Design/os-mockups/-/raw/master/portals/portals.png) have the following wording:

```
Allow %s to Access Location?
```

File: src/location.c [hunk 6]
```diff
@@ -545,11 +545,11 @@ handle_start_in_thread_func (GTask *task,
...
           if (g_desktop_app_info_has_key (info, "X-Geoclue-Reason"))
             subtitle = g_desktop_app_info_get_string (info, "X-Geoclue-Reason");
           else
-            subtitle = g_strdup_printf (_("%s wants to use your location."), name);
+            subtitle = g_strdup_printf (_("%s wants to know the location of this device."), name);
         }
 
       body = _("Location access can be changed at any time from the privacy settings.");
```
Comment (AlessandroBono) [inline src/location.c:555]: According to the mockups this should be replaced with the content of the subtitle and the subtitle should not exists.

File: src/location.c [hunk 7]
```diff
@@ -532,8 +532,8 @@ handle_start_in_thread_func (GTask *task,
 
       if (g_str_equal (app_id, ""))
         {
-          title = g_strdup (_("Grant Access to Your Location?"));
-          subtitle = g_strdup (_("An application wants to use your location."));
+          title = g_strdup (_("Allow Access to Your Location?"));
```
Comment (AlessandroBono) [inline src/location.c:535]: The latest [mockups](https://gitlab.gnome.org/Teams/Design/os-mockups/-/raw/master/portals/portals.png) dropped the `Your` word. We should drop it here too.

File: po/cs.po (hunk 4 above)
Comment (matthiasclasen) [reply po/cs.po:125]: "wants to known" should be "wants to know"

But I disagree that this is clearer. Also, it is longer.

File: po/cs.po [hunk 8]
```diff
@@ -116,21 +116,23 @@ msgid "Grant Access"
...
 msgstr "Nějaká aplikace chce použít vaši polohu."
 
 #: src/location.c:548
-#, c-format
-msgid "Give %s Access to Your Location?"
+#, fuzzy, c-format
+msgid "Allow %s to Access to Your Location?"
```
Comment (matthiasclasen) [inline po/cs.po:130]: Nope, the 'to' makes it incorrect English.

Comment (allanday) [reply po/cs.po:130]: Note that that's the second "to" that isn't needed, not the first one!

For example, "Allow Maps to Access Your Location".

**Comment by GeorgesStavracas** [comment]
@0xMRTT please update as per last review comments

**Comment by ebassi** [comment]
Superseded by #951

**Pull request closed without merging**